### PR TITLE
Hardening the library

### PR DESCRIPTION
**Pull request by visweshwar**
Externalizing Properties , adding logging and quashing a few bugs

### CODE REVIEW

File: src/com/telesign/phoneid/PhoneId.java [hunk 1]
```diff
@@ -24,176 +24,182 @@
...
+     *                    (available from the TeleSign Client Portal).
+     */
+    public PhoneId(String customer_id, String secret_key) {
+
+        this.customer_id = customer_id;
+        this.secret_key = secret_key;
+    }
+
+    public String getTelesign_url() {
+        return telesign_url == null ? "https://rest.telesign.com" : telesign_url;
```
Comment (gibffe) [inline src/com/telesign/phoneid/PhoneId.java]: Pointless. Just assign default value up front. Setter can always overwrite it.

File: src/com/telesign/util/TeleSignRequest.java [hunk 2]
```diff
@@ -29,353 +29,381 @@
...
-	}
-
-	/**
-	 * Returns the set of both the standard <u>and</u> the
-	 * <em>TeleSign-specific</em> Request header fields.
-	 *
-	 * @return A sorted key/value mapping that contains all of the Request header fields.
-	 */
-	public Map<String, String> getAllHeaders() {
+    private static Logger logger = Logger.getLogger(TeleSignRequest.class);
```
Comment (gibffe) [inline src/com/telesign/util/TeleSignRequest.java]: Github diffs are useless if you touch every single whitespace in the file - hard to see what changed :)

File: src/com/telesign/util/TeleSignRequest.java [hunk 3]
```diff
@@ -29,353 +29,381 @@
...
+            int i = 0;
 
-		String signingString = getSigningString(customer_id);
-		String signature;
-		String url_output = "";
+            for (String key : params.keySet()) {
 
-		// Create the absolute form of the resource URI, and place it in a string buffer.
-		StringBuffer full_url = new StringBuffer(base).append(resource);
+                if (++i != 0) {
```
Comment (gibffe) [inline src/com/telesign/util/TeleSignRequest.java]: If you're resorting to using indexes, just fall back to standard for loop. This code stinks and is obviously wrong. Example url it produces: 

https://rest.telesign.com/v1/phoneid/standard/15551234567?&code=001

?&  <-- ?!!

File: test/com/telesign/VerifyTest.java [hunk 4]
```diff
@@ -23,184 +24,168 @@
...
+        VerifyResponse ret = ver.sms(PHONE_NUMBER);
+        assertNotNull(ret);
+        assertTrue(ret.errors.length == 0);
+
+        String reference_id = ret.reference_id;
+
+        VerifyResponse ret2 = ver.status(reference_id);
+        assertNotNull(ret2);
+        assertTrue(ret2.errors.length == 0);
+    }
```
Comment (gibffe) [inline test/com/telesign/VerifyTest.java]: Really hard to see what changed when you delete/replace big blocks of code.

**Comment by hmorales** [comment]
Agree with @gibffe points, especially the difficulty in reviewing diffs when the entire file was modified.

**Pull request closed without merging**